Source/WebCore/loader/PingLoader.cpp

```diff
@@ -146,9 +146,15 @@ void PingLoader::sendPing(Frame& frame, const URL& pingURL, const URL& destinati
     FrameLoader::addHTTPOriginIfNeeded(request, SecurityPolicy::generateOriginHeader(document.referrerPolicy(), request.url(), sourceOrigin));
 
     frame.loader().updateRequestAndAddExtraFields(request, IsMainResource::No);
+
+    // https://www.w3.org/TR/2009/WD-html5-20090423/history.html#hyperlink-auditing
     request.setHTTPHeaderField(HTTPHeaderName::PingTo, destinationURL.string());
-    if (!SecurityPolicy::shouldHideReferrer(pingURL, frame.loader().outgoingReferrer()))
-        request.setHTTPHeaderField(HTTPHeaderName::PingFrom, document.url().string());
+    if (document.securityOrigin().isSameOriginAs(SecurityOrigin::create(pingURL).get()))
+        request.setHTTPHeaderField(HTTPHeaderName::PingFrom, document.url().strippedForUseAsReferrer());
+    else if (!document.url().protocolIs("https"_s)) {
+        request.setHTTPHeaderField(HTTPHeaderName::PingFrom, document.url().strippedForUseAsReferrer());
+        request.setHTTPReferrer(document.url().strippedForUseAsReferrer());
```

Comment:
Spec seems to say `no-referrer`, yet you include a referrer here? Why is that?

Reply:
The old link said to.  That link was great feedback.